src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -39,104 +41,101 @@ public void Dispose()
         {
             _senderCts.Cancel();
             _pollCts.Cancel();
-            _pipeline?.Dispose();
         }
 
-        public Task StartAsync(Uri url, IPipelineConnection pipeline)
+        public Task StartAsync(Uri url, IChannelConnection<Message> toFromConnection)
         {
-            _pipeline = pipeline;
-
-            // Schedule shutdown of the poller when the output is closed
-            pipeline.Output.Writing.ContinueWith(_ =>
-            {
-                _pollCts.Cancel();
-                return TaskCache.CompletedTask;
-            });
+            _toFromConnection = toFromConnection;
 
             // Start sending and polling
-            _poller = Poll(Utils.AppendPath(url, "poll"), _pollCts.Token);
-            _sender = SendMessages(Utils.AppendPath(url, "send"), _senderCts.Token);
+            _poller = Poll(Utils.AppendPath(url, "poll"), _pollCts.Token).ContinueWith(_ => _senderCts.Cancel());
```

Comment:
The problem is that even if you TryComplete from outside the send loop will continue to run until it drains the channel. This means that until the channel is drained we will keep sending requests to the server even though the connection is already disconnected. Cancellation token prevents that. 
(Note: the channel from application to transport is not currently drained - I think it should be done by the side writing to the channel i.e. HubConnection in case of hubs or the user himself when using Sockets but this is one of the issues I wanted to talk about)
I also think that we actually could use just one cancellation tokens for both send and poll loops.